addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -993,6 +995,15 @@ const ColorPaletteWidget = Widget.extend({
         this._updateGradient();
         this._activateGradientSlider($(this.pickers['custom_gradient'].querySelector('.o_slider_multi input')));
     },
+    /**
+     * @private
+     */
+    _onUpdateColorButtons() {
+        this.el.querySelectorAll('button[data-color].o-color-button').forEach(elem => {
+            const colorName = elem.dataset.color;
+            elem.style.backgroundColor = weUtils.getCSSVariableValue(colorName);
+        });
+    },
```

Comment:
> when you have time could you check [this commit](https://github.com/odoo-dev/odoo/pull/1149/commits/0e285399daf63b8e368a55fe81b7e950c46157e6) (the last commit). It's related to the color palettes and the reloading of assets in the backend. Now, I add an inline style for the o-colors. Unfortunately I need to find a way to update these colors when the presets gets changed. I ended up using the event system to pass the message to the we-colorpicker and then to the ColorPalette widget. What do you think of this?

@detrouxdev The principle seems ok, although maybe I would simplify even furthur and just ask the colorpicker to update their colorpalette colors each time they are opened, no matter if a change of bundle occured before or not. I think the cost in performance would be negligeable compared to the gain in code readability (to test).

Actually, in your current code, I am not sure it actually works: how are the colorpickers of existing snippet options updated? I understand the ones of the "ThemeColors" option are updated but not the other ones?

=> Actually, there could be another possibility which is worth trying:
- At the start of the editor, "write a CSS variable on the snippet menu itself with the colors that you need. So something like:
```javacript
snippetMenuEl.style.setProperty('--o-color-1', weUtils.getCSSVariableValue(--o-color-1));
```
- The colorpicker buttons can then just be created with:
```
elem.style.backgroundColor = 'var(--o-color-1)';
```
- And then you only need to update the `snippetMenuEl` CSS variables when the bundles (frontend only) are reloaded (you can just trigger_up an event when that occurs that is catched by the snippet menu instance.

=> Advantage: we do not reload assets_backend (relying on potentially-false-or-there-by-mistake variable on the backend HTML element), we never have to update all the colorpicker button as their style depends only on CSS variable declared on the editor panel itself, we only need to update each color once (not once per colorpicker) at the exact time the bundle has been reloaded.

Reply:
@qsm-odoo I updated the commit => WIP Color Platte using CSS Variable

I didn't use it for the o_cc color preview elements (The ones with the title, buttons and such) because it wouldn't work with buttons since their text color isn't defined by a variable but rather by a SCSS rule, AFAIK. If you're ok with the rest of the changes, I'll rename / squash the commit.